opentelemetry/api/trace/__init__.py

```diff
@@ -0,0 +1,190 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+The OpenTelemetry tracing API describes the classes used to generate
+distributed traces.
+
+The :class:`.Tracer` class controls access to the execution context, and
+manages span creation. Each operation in a trace is represented by a
+:class:`.Span`, which records the start, end time, and metadata associated with
+the operation.
+
+This module provides abstract (i.e. unimplemented) classes required for
+tracing, and a concrete no-op ``BlankSpan`` that allows applications to use the
+API package alone without a supporting implementation.
+
+The tracer supports implicit and explicit context propagation. By default spans
+are created as children of the currently active span, and the newly-created
+span becomes the new active span::
+
+    from opentelemetry.sdk.trace import tracer
+
+    # Create a new root span, set it as the current span in context
+    with tracer.span("parent"):
+        # Attach a new child and update the current span
+        with tracer.span("child"):
+            do_work():
+        # Close child span, set parent as current
+    # Close parent span, set default span as current
+
+Under explicit context propagation there is no concept of an active span, and
+the caller is responsible for managing the span's lifetime::
+
+    from opentelemetry.sdk.trace import tracer
+    from your.integration import deserialize_span
+
+    parent = deserialize_span(serialized_span)
+    # Explicit parent span assignment
+    with tracer.span("child", parent=parent) as child:
+        do_work(span=child)
+
+Applications should generally use a single global tracer, and use either
+implicit or explicit context propagation consistently throughout.
+
+.. versionadded:: 0.1.0
+"""
+
+from __future__ import annotations
+
+from typing import Iterator
+from contextlib import contextmanager
+
+
+class Tracer(object):
+    """Handles span creation and in-process context propagation.
```

Comment:
We use imperative like "Return the current span" in the method descriptions. Here we use a descriptive statement. We should decide on either one or the other (I'd prefer descriptive tone like here, since I expect most readers of the docs wanting to instrument their code instead of wanting to create a new implementation for the interface.

Reply:
Descriptive SGTM.